memCache/entryCache.go

```diff
@@ -433,3 +215,216 @@ func (e *Entry) timeToUpdate() bool {
 	}
 	return false
 }
+
+type EntryCache struct {
+	log            *log.Logger
+	clk            clock.Clock
+	requestTimeout time.Duration
+	entries        map[string]*Entry   // one-to-one map keyed on name -> entry
+	lookupMap      map[[32]byte]*Entry // many-to-one map keyed on sha256 hashed OCSP requests -> entry
```

Comment:
Or another way of putting it: Why do you need a map from name -> entry? AFAICT it's only used for removing entries, which you could probably do another way.